src/ngx_http_lua_socket_tcp.c

```diff
@@ -6395,4 +6392,112 @@ ngx_http_lua_ffi_socket_tcp_del_udata(ngx_http_lua_socket_tcp_upstream_t *u,
 }
 
 
+int
+ngx_http_lua_ffi_socket_tcp_getoption(ngx_http_lua_socket_tcp_upstream_t *u,
+    int option, int *val, u_char *err, size_t *errlen)
+{
+    u_char    errstr[NGX_MAX_ERROR_STR];
+    socklen_t len;
+    u_char   *p;
+    int       fd, r;
```

Comment:
We usually use `rc` instead of `r` for such variables.

Reply:
accept